### ISSUE TITLE
Combo box navigation/focus issues with SR 

### ISSUE THREAD

**Issue by samsadasivan** [cds-snc/platform-forms-client#5835]
**_Specs: Windows
Screen Readers Used: JAWS, NVDA
Browsers Tested: Google Chrome, Microsoft Edge, Internet Explorer_**

**Same issue is observed on 2 pages.** 

1. Applicant Information 
2. Additional Information - Funding Support 

Component: 
Combo Box – “Department or agency” and “Select the type of applicaiton:” 

**_Observation:_** 

When using a screen reader with the “Department or agency” combo box: 

1. The screen reader does not announce how to select a list item. When the combo box is focused and a user navigates the dropdown list using arrow keys, the screen reader does not provide any guidance (e.g., “Press Enter to select this item”). 

2. After selecting an item, the screen reader does not confirm that an item has been selected. After a user selects a department or agency from the list, there is no audible feedback or announcement from the screen reader confirming the selection.

**Comment by thiessenp-cds** [comment]
For 1) I added more semantics to show the combobox has a popup with a list. Adding more keyboard behaviour instructions would probably be redundant since the the combobox has a role="combobox" and an AT user is probably familiar with how comboboxes work.
For 2) The combobox selection is announced (e.g. below). It's possible a code update fixed this in the mean time.
<img width="1057" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK1c5Q2c9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--21b1823f14810a708a2834f2bfe0ffe9027a0521/Screenshot%202025-07-16%20at%2011.40.34%E2%80%AFAM.png" alt="Screenshot 2025-07-16 at 11.40.34 AM.png" />
